x-pack/plugins/fleet/server/services/epm/packages/bulk_install_packages.ts

```diff
@@ -55,37 +61,73 @@ export async function bulkInstallPackages({
     return pkg.name;
   });
 
-  const limiter = pLimit(10);
+  const forbiddenPackages = uniquePackages
+    .filter((pkg) =>
+      typeof pkg === 'string'
+        ? FORBIDDEN_BULK_INSTALL_PACKAGE_NAMES.includes(pkg)
+        : FORBIDDEN_BULK_INSTALL_PACKAGE_NAMES.includes(pkg.name)
+    )
+    .map((pkg) => (typeof pkg === 'string' ? pkg : pkg.name));
+
+  if (forbiddenPackages.length > 0) {
+    logger.warn(
+      `Package(s) cannot be bulk installed and will be ignored: ${forbiddenPackages.join(',')}`
+    );
+  }
+
+  const filteredPackagesToInstall = uniquePackages.filter(
+    (pkg) => !forbiddenPackages.includes(typeof pkg === 'string' ? pkg : pkg.name)
+  );
+
+  const maxConcurrentInstalls =
+    appContextService.getConfig()?.internal?.maxConcurrentBulkInstallations ??
+    DEFAULT_MAX_CONCURRENT_INSTALLS;
+
+  const limiter = pLimit(maxConcurrentInstalls);
 
-  const packagesResults = await Promise.allSettled(
-    uniquePackages.map(async (pkg) => {
+  const packagesResults = await Promise.allSettled<{
+    name: string;
+    version: string;
+    prerelease?: boolean;
+    skipDataStreamRollover?: boolean;
+    forbidden?: boolean;
+  }>(
+    filteredPackagesToInstall.map(async (pkg) => {
       return limiter(async () => {
+        const packageName = typeof pkg === 'string' ? pkg : pkg.name;
+        const isPackageForbidden = FORBIDDEN_BULK_INSTALL_PACKAGE_NAMES.includes(packageName);
+
+        if (isPackageForbidden) {
+          logger.warn(`Package cannot be bulk installed and will be ignored: ${packageName}`);
+
+          return Promise.resolve({
```

Comment:
Is it enough to just log a warning? We want to ignore this package install but the user won't know, unless they look at the logs.

Reply:
I've updated this so that the response looks like this 

```json
{
  "items": [
    {
      "name": "security_detection_engine",
      "statusCode": 422,
      "error": "Bulk installation of security_detection_engine is forbidden. Please install it via the single package installation API."
    }
  ],
  "response": [
    {
      "name": "security_detection_engine",
      "statusCode": 422,
      "error": "Bulk installation of security_detection_engine is forbidden. Please install it via the single package installation API."
    }
  ]
}
```